File: solutions/cpp/pangram/pangram.cpp
#include "pangram.h"

#include <set>

namespace pangram {

    auto is_pangram(std::string s) -> bool
    {
        std::set<char> letters;

        for (unsigned char c : s)
        {
            if (std::isalpha(c))
            {
                letters.insert(std::tolower(c));
            }
        }

        return letters.size() == 26;
    }

}  // namespace pangram

File: solutions/haskell/pangram/src/Pangram.hs
module Pangram (isPangram) where

import qualified Data.Set as Set
import qualified Data.Char as Char

isPangram :: String -> Bool
isPangram text = Set.size (Set.fromList (map Char.toLower (filter Char.isAlpha (filter  Char.isAscii text)))) == 26
